When put null value into cache, java.lang.NullPointerException got thrown

Putting a null value into the caffeine cache caused a NullPointerException, the root cause it Caffeine doesn't accept null value.

```java
Caused by: java.lang.NullPointerException
	at java.base/java.util.Objects.requireNonNull(Objects.java:221)
	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.put(UnboundedLocalCache.java:372)
	at com.github.benmanes.caffeine.cache.UnboundedLocalCache.put(UnboundedLocalCache.java:367)
	at com.github.benmanes.caffeine.cache.LocalManualCache.put(LocalManualCache.java:130)
	at org.mybatis.caches.caffeine.CaffeineCache.putObject(CaffeineCache.java:57)
	at [=============hidden==========].cache.CacheDelegator.putObject(CacheDelegator.java:42)
	at org.apache.ibatis.cache.decorators.LoggingCache.putObject(LoggingCache.java:49)
	at org.apache.ibatis.cache.decorators.TransactionalCache.flushPendingEntries(TransactionalCache.java:120)
	at org.apache.ibatis.cache.decorators.TransactionalCache.commit(TransactionalCache.java:99)
	at org.apache.ibatis.cache.TransactionalCacheManager.commit(TransactionalCacheManager.java:45)
	at org.apache.ibatis.executor.CachingExecutor.commit(CachingExecutor.java:120)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.ibatis.plugin.Plugin.invoke(Plugin.java:64)
	at com.sun.proxy.$Proxy119.commit(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.apache.ibatis.plugin.Invocation.proceed(Invocation.java:49)
	at tansun.creditx.audit.DataChangeInterceptor.intercept(DataChangeInterceptor.java:94)
	at org.apache.ibatis.plugin.Plugin.invoke(Plugin.java:62)
	at com.sun.proxy.$Proxy119.commit(Unknown Source)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.commit(DefaultSqlSession.java:220)
```
where TransactionalCache.java:120 is:
```
  private void flushPendingEntries() {
    for (Map.Entry<Object, Object> entry : entriesToAddOnCommit.entrySet()) {
      delegate.putObject(entry.getKey(), entry.getValue());
    }
    for (Object entry : entriesMissedInCache) {
      if (!entriesToAddOnCommit.containsKey(entry)) {
        delegate.putObject(entry, null);     <============================                    
      }
    }
  }
```

--- a/src/main/java/org/mybatis/caches/caffeine/CaffeineCache.java
+++ b/src/main/java/org/mybatis/caches/caffeine/CaffeineCache.java
@@ -1,5 +1,5 @@
 /*
- *    Copyright 2016-2022 the original author or authors.
+ *    Copyright 2016-2023 the original author or authors.
  *
  *    Licensed under the Apache License, Version 2.0 (the "License");
  *    you may not use this file except in compliance with the License.
@@ -54,7 +54,10 @@ public final class CaffeineCache implements Cache {
 
   @Override
   public void putObject(Object key, Object value) {
-    this.cache.put(key, value);
+    // Do not allow null values to be cached as not allowed in caffeine cache
+    if (key != null && value != null) {
+      this.cache.put(key, value);
+    }
   }
 
   @Override

